spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/direct/BigQueryRDDContext.java

```diff
@@ -76,6 +76,9 @@ public BigQueryRDDContext(
     this.options = options;
     this.bqSchema = bqSchema;
     this.streamNames = BigQueryUtil.getStreamNames(readSession);
+    if (bigQueryClientFactory != null) {
```

Comment:
Why do we need the null checking?

Reply:
There are some tests which sends bigQueryReadClientFactory as `null` so this causes a NPE in those tests.